hadoop-ozone/interface-client/pom.xml

```diff
@@ -51,16 +71,46 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
         <extensions>true</extensions>
         <executions>
           <execution>
-            <id>compile-protoc</id>
+            <id>compile-protoc-OmGrpc</id>
             <goals>
               <goal>compile</goal>
               <goal>test-compile</goal>
+              <goal>compile-custom</goal>
+              <goal>test-compile-custom</goal>
             </goals>
             <configuration>
+              <protocArtifact>
+                com.google.protobuf:protoc:${protobuf.version}:exe:${os.detected.classifier}
+              </protocArtifact>
               <protoSourceRoot>${basedir}/src/main/proto/</protoSourceRoot>
+              <includes>
+                <include>OmClientProtocol.proto</include>
+              </includes>
+              <outputDirectory>target/generated-sources/protobuf/java</outputDirectory>
+              <clearOutputDirectory>false</clearOutputDirectory>
+              <pluginId>grpc-java</pluginId>
+              <pluginArtifact>
+                io.grpc:protoc-gen-grpc-java:${grpc-compile.version}:exe:${os.detected.classifier}
+              </pluginArtifact>
+            </configuration>
+          </execution>
+          <execution>
```

Comment:
To reduce duplication between the two executions, I suggest:

 * moving common configuration outside `<executions>` into a plugin-level `<configuration>`
 * separating `protobuf` and `grpc` compilation:
   * let `compile-protoc-2` compile all proto files
   * let `compile-OmGrpc` compile OM proto only with `grpc-java` plugin

Something like this:

```xml
      <plugin>
        <groupId>org.xolstice.maven.plugins</groupId>
        <artifactId>protobuf-maven-plugin</artifactId>
        <version>${protobuf-maven-plugin.version}</version>
        <extensions>true</extensions>
        <configuration>
          <protoSourceRoot>${basedir}/src/main/proto/</protoSourceRoot>
          <outputDirectory>target/generated-sources/java</outputDirectory>
          <clearOutputDirectory>false</clearOutputDirectory>
        </configuration>
        <executions>
          <execution>
            <id>compile-protoc-2</id>
            <goals>
              <goal>compile</goal>
              <goal>test-compile</goal>
            </goals>
            <configuration>
              <protocArtifact>
                com.google.protobuf:protoc:${protobuf.version}:exe:${os.detected.classifier}
              </protocArtifact>
            </configuration>
          </execution>
          <execution>
            <id>compile-grpc</id>
            <goals>
              <goal>compile-custom</goal>
              <goal>test-compile-custom</goal>
            </goals>
            <configuration>
              <includes>
                <include>OmClientProtocol.proto</include>
              </includes>
              <pluginId>grpc-java</pluginId>
              <pluginArtifact>
                io.grpc:protoc-gen-grpc-java:${io.grpc.version}:exe:${os.detected.classifier}
              </pluginArtifact>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

Reply:
Thanks for uncovering the duplication in executions.  We can also rewrite the two executions into a single execution by having the _`compile-OmGrpc`_ execution id also generate the protoc generated code that id _`compile-protoc-2`_ currently compiles, namely the `<include>Security.proto</include>`.  In this case the single execution is : 
```
          <execution>
            <id>compile-protoc-OmGrpc</id>
            <goals>
              <goal>compile</goal>
              <goal>test-compile</goal>
              <goal>compile-custom</goal>
              <goal>test-compile-custom</goal>
            </goals>
            <configuration>
              <protocArtifact>
                com.google.protobuf:protoc:${protobuf.version}:exe:${os.detected.classifier}
              </protocArtifact>
              <protoSourceRoot>${basedir}/src/main/proto/</protoSourceRoot>
              <includes>
                <include>OmClientProtocol.proto</include>
                <include>Security.proto</include>
              </includes>
              <outputDirectory>target/generated-sources/protobuf/java</outputDirectory>
              <clearOutputDirectory>false</clearOutputDirectory>
              <pluginId>grpc-java</pluginId>
              <pluginArtifact>
                io.grpc:protoc-gen-grpc-java:${grpc-compile.version}:exe:${os.detected.classifier}
              </pluginArtifact>
            </configuration>
          </execution>
```
Either is fine, spit the two or have a single execution - what are your thoughts on this?

Reply:
Depends on whether you need GRPC generated code for `Security.proto`, I guess.